File: programmers/crain.py
def solution(board, moves):
  basket = []
  score = 0
  for move in moves:
    idx = move - 1
    for row in board:
      if row[idx] == 0:
        continue
      basket.append(row[idx])
      row[idx] = 0
      break
    if len(basket) >= 2 and basket[len(basket) - 1] == basket[len(basket) - 2]:
      basket.pop()
      basket.pop()
      score += 2
  return score

print(solution([[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]], [1,5,3,5,1,2,1,4]))

File: programmers/crain.java
package algoexam;

import java.util.ArrayList;
import java.util.Stack;

public class Crain {
    public int solution(int[][] board, int[] moves) {
        ArrayList<Integer> basket = new ArrayList<>();
        int score = 0;
        for(int move : moves) {
            int idx = move - 1;
            for(int[] row : board){
                if(row[idx] != 0){
                    basket.add(row[idx]);
                    row[idx] = 0;
                    break;
                }
            }
            int basketSize = basket.size();
            if(basketSize >= 2 && basket.get(basketSize - 1) == basket.get(basketSize - 2)) {
                basket.remove(basketSize - 1);
                basket.remove(basketSize - 2);
                score +=2 ;
            }
        }
        Stack<Integer> bas = new Stack<>();
        bas.add(1);
        bas.add(2);
        bas.add(3);
        bas.get(3);
        
        return score;
    }
}
